src/helpers/request.ts

```diff
@@ -0,0 +1,85 @@
+import { ParsedUrlQueryInput } from 'querystring';
+import { addQueryToUrl, appendPathnameToUrl } from 'url-transformers';
+import { compactDefined, flow } from './fp';
+import { ApiResponse, handleFetchResponse, HandleResponse } from './response';
+import { isDefined, OmitStrict } from './typescript';
+
+type BuildUrlParams = {
+  pathname: string;
+  query?: ParsedUrlQueryInput;
+};
+
+const buildUrl = ({ pathname, query = {} }: BuildUrlParams) =>
+  flow(appendPathnameToUrl(pathname), addQueryToUrl(compactDefined(query)));
+
+type FetchParams = Pick<RequestInit, 'method'>;
+/**
+ * The params generated by the library
+ */
+type BaseRequestParams = BuildUrlParams & FetchParams;
+
+/**
+ * Additional fetch options provided by the user on a per-call basis
+ */
+type AdditionalPerFetchParams = Omit<RequestInit, keyof FetchParams>;
+type CompleteRequestParams = BaseRequestParams & AdditionalPerFetchParams;
+
+/**
+ * helper used to type-check the arguments, and add default params for all requests
+ */
+export const createRequestParams = <Args extends {}>(fn: (a: Args) => BaseRequestParams) => (
```

Comment:
`createRequestHandler` might be a better name for this, for consistency with the `HandleRequest` type I suggested in another comment.